vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt

```diff
@@ -69,7 +71,7 @@ class ScreenRecorderTestRunListener(
                 recorder?.join(awaitMillis)
             }
             logger.trace { "join ${join}ms" }
-            if (hasFailed) {
+            if (hasFailed || screenRecordingPolicy == ScreenRecordingPolicy.ON_ANY) {
```

Comment:
```hasFailed``` is subset of `ON_ANY`, will be better to swap the condition

Reply:
will update